### PR TITLE
Fix serialization issues

### PR DESCRIPTION
**Pull request by Tim-Linschoten**

### CODE REVIEW

File: core/src/main/scala/com/ing/baker/compiler/RecipeValidations.scala [hunk 1]
```diff
@@ -136,9 +136,9 @@ object RecipeValidations {
       .map(c => s"Event class: $c is not serializable by akka")
 
     val ingredientSerializationErrors: Seq[String] =
-      compiledRecipe.ingredients.values
-        .filterNot(hasAkkaSerializer)
-        .map(c => s"Ingredient class: $c is not serializable by akka").toSeq
+      compiledRecipe.ingredients
+        .filterNot{case (c, v) => hasAkkaSerializer(v) }
```
Comment (adilakhter) [inline core/src/main/scala/com/ing/baker/compiler/RecipeValidations.scala]: - `c` is not used anywhere. consider using `_`
-  could you please add spaces before and after the `{`

File: core/src/main/scala/com/ing/baker/java_api/JBaker.scala [hunk 2]
```diff
@@ -91,10 +91,8 @@ class JBaker private (jRecipe: JRecipe,
     *
     * @param processId the identifier of the process
     * @param event The event to fire
-    * @throws BakerException
     * @return
     */
-  @throws(classOf[BakerException])
```
Comment (adilakhter) [inline core/src/main/scala/com/ing/baker/java_api/JBaker.scala]: Correct me if I am wrong?  

We might get the following exceptions, right?

- CancellationException
- ExecutionException
-  InterruptedException
- TimeoutException

Comment (Tim-Linschoten) [inline core/src/main/scala/com/ing/baker/java_api/JBaker.scala]: Those are all runtime exceptions.
We only need to supply all checked exceptions and only for the classes directly used in the Java code.

Comment (adilakhter) [inline core/src/main/scala/com/ing/baker/java_api/JBaker.scala]: aha..okay!

**Comment by adilakhter** [comment]
LGTM! ;)

**Pull request merged**